src/components/views/Sidebar.js

```diff
@@ -0,0 +1,361 @@
+                minHeight: 48,
+                justifyContent: open ? 'initial' : 'center',
+                px: 2.5,
+            }}
+            onClick={onClick}
+        >
+            <ListItemIcon
+                sx={{
+                    minWidth: 0,
+                    mr: open ? 3 : 'auto',
+                    justifyContent: 'center',
+                }}
+            >
+                {icon}
+            </ListItemIcon>
+
+            <ListItemText primary={text} sx={{ opacity: open ? 1 : 0 }} />
+        </ListItemButton>
+      </Tooltip>
+    );
+};
+
+
+// ==================================================
+// sidebar and header
+export default function Sidebar({view, children}) {
+    const theme = useTheme();
+    const [open, setOpen] = React.useState(false);
+    const history = useHistory();
+
+    // ==================================================
+    // handling drawer state (open/closed)
+    const handleDrawerOpen = () => {
+        setOpen(true);
+    };
+
+    const handleDrawerClose = () => {
+        setOpen(false);
+    };
+
+
+    // ==================================================
+    // functions for navigation
+    async function logout() {
+        try {
+            const requestBody = ""
+
+            await api.put(`/users/logout/${localStorage.getItem('id')}`,
+                requestBody);
+
+        } catch (error) {
+            alert(`Something went wrong during the logout: \n${handleError(error)}`);
+        }
+        localStorage.removeItem('token');
+        localStorage.removeItem('id');
+        history.push('/login');
+    }
+
+
+    function goToUserPage() {
+        try {
+            history.push(`/users/${localStorage.getItem("id")}`)
+        }catch (error) {
+            alert(`Something went wrong while trying to access your profile: \n${handleError(error)}`);
+        }
+    }
+
+
+    // ============================================================
+    // drawer components that differ in their listed sidebar items (depending on the view prop)
+    const LoginDrawer = () => {
+        return (
+            <Drawer variant="permanent" open={open}>
+
+            {/* header of the sidebar */}
+                <DrawerHeader>
+                    <IconButton onClick={handleDrawerClose}>
+                    {theme.direction === 'rtl' ? <ChevronRightIcon /> : <ChevronLeftIcon />}
+                    </IconButton>
+                </DrawerHeader>
+
+
+                <Divider />
+
+                {/* list of sidebar items */}
+                <List>
+                    <SidebarItem
+                        open={open}
+                        text="Login"
+                        onClick={() => {history.push('/login')}}
+                        icon={<LoginIcon />}   
+                    />
+                    <SidebarItem
+                        open={open}
+                        text="Registration"
+                        onClick={() => {history.push('/login')}}
```

Comment:
Must: This should be /registration

Reply:
> Looks very nice. Where can you adjust the styling? E.g. if we want to change the header colour In the registration, no sidebar is visible. Can you merge Chat and Matches, because we are doing both on one page. (the match overview will become the chatoverview). Otherwise, it looks great! Great work!

You can style the header/drawer with the sx prop (https://mui.com/system/the-sx-prop/). E.g.
![image](https://user-images.githubusercontent.com/52537652/167317266-030ea007-46e2-41cb-b8c5-575e798b0ff2.png)